File: cpp/src/Employee.cpp


#include "Employee.h"

#include "SeniorityLevel.h"

Employee::Employee(double annualSalary, int tenure, SeniorityLevel* seniority)
    : annualSalary(annualSalary), tenure(tenure), seniority(seniority) {}

double Employee::getAnnualSalary() const {
    return annualSalary;
}

int Employee::getTenure() const {
    return tenure;
}

SeniorityLevel* Employee::getSeniority() const {
    return seniority;
}

File: typescript/src/Employee.ts
import {SeniorityLevel} from "./SeniorityLevel";

export class Employee {
  private readonly annualSalary: number;
  private readonly tenure: number;
  private readonly seniority: SeniorityLevel;

  constructor(annualSalary: number, tenure: number, seniority: SeniorityLevel) {
    this.annualSalary = annualSalary;
    this.tenure = tenure;
    this.seniority = seniority;
  }

  getAnnualSalary(): number {
    return this.annualSalary;
  }

  getTenure(): number {
    return this.tenure;
  }

  getSeniority(): SeniorityLevel {
    return this.seniority;
  }
}
